packages/aserto/src/aserto/client/directory/aio/__init__.py

```diff
@@ -36,12 +37,44 @@
 from grpc import ChannelCredentials, StatusCode, ssl_channel_credentials
 
 
+@dataclass(frozen=True)
+class ObjectIdentifier:
+    """
+    Unique identifier of a directory object.
+    """
+
+    key: str
+    type: str
+
+
+@dataclass(frozen=True)
+class GetRelationResponse:
+    """
+    Response to get_relation calls.
+
+    Attributes
+    ----
+    relation    The returned relation.
+    objects     If with_relations is True, a mapping from "type:key" to the corresponding object.
+    """
+
+    relation: Relation
+    objects: Optional[Mapping[str, Object]]
```

Comment:
Just an idea, but it might be easier for users to consume the `objects` map if the keys were of type `ObjectIdentifier` instead of strings.
It would just be a dict comprehension that splits the string on `:` and builds an `ObjectIdentifier` with the type and key.